sql/core/src/main/scala/org/apache/spark/sql/execution/local/HashJoinNode.scala

```diff
@@ -24,19 +24,33 @@ import org.apache.spark.sql.execution.joins._
 import org.apache.spark.sql.execution.metric.SQLMetrics
 
 /**
+ * A node for inner hash equi-join. It can be used individually or wrapped by other
+ * inner hash equi-join nodes such as [[BinaryHashJoinNode]]. This node takes a already
+ * built [[HashedRelation]] and a [[LocalNode]] representing the streamed side.
+ * If this node is used individually, `isWrapped` should be set to false.
+ * If this node is wrapped in another node, `isWrapped` should be set to true
```

Comment:
As discussed offline, it might be better to put the common code in a trait instead of wrapping local nodes. Right now it's kind of confusing because sometimes we call `prepare` or `close` but sometimes we don't.